File: Coding_Platforms/CodeChef/Practice/Golang/avg_number.go
package main

import "fmt"

func main() {

	//create a veriable for test cases and take input
	var test_case int
	fmt.Scanln(&test_case)

	for i := 0; i < test_case; i++ { //iterate for testcases and print result

		fmt.Println(AvgNumber())
	}
}

func AvgNumber() int { //check average number

	//create variable for n, k, v, data and result and take input for n, k, and v.
	var n, k, v, data, result int
	fmt.Scanln(&n, &k, &v)

	for j := 0; j < n; j++ { //iterate for n times to take value of n.

		//create variable and take value for variable and add that value in data
		var val int
		fmt.Scan(&val)
		data += val
	}

	V := (n + k) * v //store sum of all data into V.

	if ((V - data) % k) == 0 { //if result is not integer

		if result = (V - data) / k; result > 0 { //store average number into result if average is greater then 0 then return result
			return result
		}
	}
	return -1
}

File: Coding_Platforms/CodeChef/Practice/Python/avg_number.py
def AvgNumber(nkv): # check the average number

    # take sum of all data into and
    # subtract the sum of second input list
    data = ((nkv[0]+nkv[1])*nkv[2])-sum(list(map(int, input().split(" ")))) 
    
    # if result is positive integer 
    if ((data)%nkv[1]) == 0 and data > 0: 
        return data//nkv[1]
    
    else: # else retunrn -1
        return -1

for i in range(int(input())): # iterate over testcases
    
    # take list as input and pass into AvgNumber and print result
    print(AvgNumber(list(map(int, input().split(" ")))))